backend/src/DataSources/users.spec.js

```diff
@@ -0,0 +1,270 @@
+const Server = require("../server");
+const bcrypt = require("bcrypt");
+const utils = require("../utils");
+const jwt = require("jsonwebtoken");
+
+let postsMemory = new PostsDataSource();
+let usersMemory = new UsersDataSource();
+let decoded
+
+beforeEach(() => {
+    decoded = {jwt}
+})
+
+const testContext = () => decoded
+
+const server = new Server({context: testContext, dataSources: ()=> ({postsDataSrc: postsMemory, usersDataSrc:usersMemory})});
+const {query, mutate } = createTestClient(server);
+
+describe("queries", () => {
+
+    describe("USERS", () => {
+        
+        const GET_USERS = gql`
+                query {
+                    users {
+                        id
+                        name
+                        email
+                    }
+                }
+            `;
+        
+        it("returns empty array", async () => {
+            usersMemory.users = []
+
+            await expect(query({query: GET_USERS}))
+            .resolves
+            .toMatchObject({
+                errors: undefined,
+                data: { users: [] }
+            })
+        })
+
+        it("given users in the database", async () => {
+            usersMemory.users = [new User({name:"Andrej", email:"andrej@gmail.com", password:"12345678"})]
+            
+            await expect(query({query: GET_USERS}))
+            .resolves
+            .toMatchObject({
+                errors: undefined,
+                data: { users: [{id:expect.anything(String), name:"Andrej", email:"andrej@gmail.com"}]}
+            })
+        })
+
+        it("indefinitely nestable query", async () => {
+            usersMemory.users = [...utils.defaultUsers];
+            postsMemory.posts = [...utils.defaultPosts];
+
+            const NESTABLE_QUERY_USERS = gql`
+                query {
+                    users {
+                        id
+                        name
+                        email
+                        posts {
+                            title
+                            author {
+                                name
+                            }
+                        }
+                    }
+                }`;
+            
+            await expect(query({query: NESTABLE_QUERY_USERS}))
+           .resolves
+           .toMatchObject({
+               errors: undefined,
+               data :{ users:
+                [
+                    {
+                        id: expect.anything(String),
+                        name: "An",
+                        email: "an@gmail.com",
+                        posts: [
+                            {
+                                title: "Just",
+                                author: {
+                                    name: "An",
+                                }
+                            }
+                        ]
+                    },
+                    {
+                        id: expect.anything(String),
```

Comment:
```suggestion
                        id: expect.any(String),
```
It's `expect.anything()` or `expect.any(Constructor)`: https://jestjs.io/docs/en/expect